app/controllers/application_controller.rb

```diff
@@ -1,3 +1,12 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery
+  include SessionsHelper
+
+protected
+  def authenticate
+  	if !current_user
```

Comment:
The indentation is way too off now, this should be 2 spaces not 4. I assume it might be because your editor does not translate tabs to spaces. I highly recommend fixing this.